### PR TITLE
Add visibility specification based on group and user URNs

### PR DESCRIPTION
**Pull request by NathanPruyne**
Modifies `set_visibility` to accept UUIDs associated with Globus Auth groups and user UUIDs via `users` and `groups` optional arguments. Without either argument the visibility will be set to public (though users desiring to make their datasets public would likely not use set_visibility). User and group UUIDs are then converted to the corresponding URNs and added to the model metadata.

Schemas have already been updated to accommodate URNs.

### CODE REVIEW

File: dlhub_sdk/models/__init__.py [hunk 1]
```diff
@@ -204,15 +204,24 @@ def set_domains(self, domains):
         self._output["dlhub"]["domains"] = domains
         return self
 
-    def set_visibility(self, visible_to):
-        """Define the list of people and groups who have permissions to see and use this model.
+    def set_visibility(self, users=[], groups=[]):
```
Comment (WardLT) [inline dlhub_sdk/models/__init__.py:207]: Mutable default arguments [can cause some weird errors](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments). Could you change these out for a tuple?

**Review by WardLT — CHANGES_REQUESTED** [review]
Looks great! Could you fix the issue I pointed out and add a unit test that adds a group/user to make sure that the resulting object validates against our JSON schema.

**Comment by NathanPruyne** [comment]
@WardLT I fixed the default arguments by replacing them with None and checking for their existence later to append to the visibility specifications. Unit test has also been added to tests/test_dlhub_client.py

**Comment by WardLT** [comment]
Thanks, @NathanPruyne !

**Pull request merged**